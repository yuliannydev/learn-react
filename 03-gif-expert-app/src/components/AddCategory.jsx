import React, { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {
    const [topic, setTopic] = useState('');
    
    const onInput = (event) => {
        setTopic(event.target.value);
    }
    
    const onSubmit = (event) => {
        event.preventDefault();
        if (topic.trim().length <= 1) return;
        
        onNewCategory(topic.trim())
        setTopic('');
    }

      return (
        <form onSubmit={onSubmit} className='search'>
            <input 
            type="text"
            placeholder='Search new gif category and push enter or botton GO'
            value={topic}
            onChange={onInput}
            />

            <button
            type='button'
            onClick={onSubmit}
            >
            Go
            </button>
        </form> 
      )
}
