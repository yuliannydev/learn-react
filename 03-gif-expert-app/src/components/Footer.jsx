import React from 'react'

export const Footer = () => {
  return (
    <footer className='footer'>
        <span>Write something in the search box</span>
        <span>Search Gif App create with Reat.js and Vanilla CSS</span>
    </footer>
  )
}
