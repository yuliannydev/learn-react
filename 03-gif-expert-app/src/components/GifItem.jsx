import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({title, image_url }) => {
  return (
    <section className='card'>
      <img src={image_url} alt={title} />
      <p>{title}</p>
    </section>
  )
} 

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  image_url: PropTypes.string.isRequired
}