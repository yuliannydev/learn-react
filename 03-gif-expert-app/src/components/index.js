export * from "./AddCategory";
export * from "./GifItem";
export * from "./ListGif";
