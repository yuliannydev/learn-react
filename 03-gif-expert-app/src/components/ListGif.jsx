import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs'
import { GifItem } from './GifItem'

export const ListGif = ({category}) => {

  const [images, setImages] = useState([])

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages)
  }

  useEffect(()=> {
    getImages();
  }, [])

  return (
    <>
      <h3> {category} </h3>
      <section className='card-grid'>
        {
          images.map((item) => (
            <GifItem 
            key={item.id}
            {...item }
            />
          ))
        }
      </section>
    </>
  )
}
