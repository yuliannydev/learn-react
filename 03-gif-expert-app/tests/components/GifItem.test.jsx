import { render } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('GifItem', () => { 

    const title = 'Hunter X Hunter';
    const image_url = 'https://hunter-x-hunter.com/h-x-h.webp';

    test('Snapshot Title and image_url required', () => { 
     const { container } = render( <GifItem title={title} image_url={image_url} /> )   
     expect(container).toMatchSnapshot();
     })
 })