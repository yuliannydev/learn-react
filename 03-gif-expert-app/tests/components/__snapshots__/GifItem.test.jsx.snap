// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Snapshot Title and image_url required 1`] = `
<div>
  <section
    class="card"
  >
    <img
      alt="Hunter X Hunter"
      src="https://hunter-x-hunter.com/h-x-h.webp"
    />
    <p>
      Hunter X Hunter
    </p>
  </section>
</div>
`;
